Helm: retrieve the created secrets by name

skitt opened the pull request:
To support Kubernetes 1.24, our Helm charts create secrets with a
fixed name; use that to retrieve the secrets instead of looking for
all annotated secrets (for Kubernetes < 1.24 there end up being two of
those).

Depends on https://github.com/submariner-io/submariner-charts/pull/237
Signed-off-by: Stephen Kitt <skitt@redhat.com>

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

skitt committed:
Helm: retrieve the created secrets by name

To support Kubernetes 1.24, our Helm charts create secrets with a
fixed name; use that to retrieve the secrets instead of looking for
all annotated secrets (for Kubernetes < 1.24 there end up being two of
those).

Signed-off-by: Stephen Kitt <skitt@redhat.com>

dfarrell07 commented on scripts/shared/lib/deploy_helm:
```diff
@@ -23,8 +23,8 @@ function setup_broker() {
-    submariner_broker_ca=$(kubectl -n "${BROKER_NAMESPACE}" get secrets -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='${BROKER_CLIENT_SA}')].data['ca\.crt']}")
-    submariner_broker_token=$(kubectl -n "${BROKER_NAMESPACE}" get secrets -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='${BROKER_CLIENT_SA}')].data.token}"|base64 --decode)
+    submariner_broker_ca=$(kubectl -n "${BROKER_NAMESPACE}" get secrets "${BROKER_NAMESPACE}-client-token" -o jsonpath="{.data['ca\.crt']}")
+    submariner_broker_token=$(kubectl -n "${BROKER_NAMESPACE}" get secrets "${BROKER_NAMESPACE}-client-token" -o jsonpath="{.data.token}"|base64 --decode)
```
This seems to be breaking even with the dependency merged.

```
[18:12:11.858] [shipyard]$ [cluster1] command kubectl --context=cluster1 -n submariner-k8s-broker get secrets submariner-k8s-broker-client-token -o jsonpath={.data['ca\.crt']}
Error from server (NotFound): secrets "submariner-k8s-broker-client-token" not found
```

liggitt commented on scripts/shared/lib/deploy_helm:
```diff
@@ -26,8 +26,8 @@ function setup_broker() {
     submariner_broker_url=$(kubectl -n default get endpoints kubernetes -o jsonpath="{.subsets[0].addresses[0].ip}:{.subsets[0].ports[?(@.name=='https')].port}")
-    submariner_broker_ca=$(kubectl -n "${BROKER_NAMESPACE}" get secrets -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='${BROKER_CLIENT_SA}')].data['ca\.crt']}")
-    submariner_broker_token=$(kubectl -n "${BROKER_NAMESPACE}" get secrets -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='${BROKER_CLIENT_SA}')].data.token}"|base64 --decode)
+    submariner_broker_ca=$(kubectl -n "${BROKER_NAMESPACE}" get secrets "${BROKER_NAMESPACE}-client-token" -o jsonpath="{.data['ca\.crt']}")
+    submariner_broker_token=$(kubectl -n "${BROKER_NAMESPACE}" get secrets "${BROKER_NAMESPACE}-client-token" -o jsonpath="{.data.token}"|base64 --decode)
```
shouldn't this be `${BROKER_CLIENT_SA}-client-token`?

skitt replied:
`${BROKER_CLIENT_SA}-token`, perhaps — that would be more legible, but the result is ultimately the same (`BROKER_CLIENT_SA` is effectively `${BROKER_NAMESPACE}-client`).

skitt commented:
https://github.com/submariner-io/submariner-charts/pull/243 should help fix this.

dfarrell07 commented:
> [submariner-io/submariner-charts#243](https://github.com/submariner-io/submariner-charts/pull/243) should help fix this.

After merge, seeing:

```
[18:28:00.299] [shipyard]$ [cluster1] command kubectl --context=cluster1 -n submariner-k8s-broker get secrets submariner-k8s-broker-client-token -o jsonpath={.data['ca\.crt']}
Error from server (NotFound): secrets "submariner-k8s-broker-client-token" not found
```

skitt commented:
> > [submariner-io/submariner-charts#243](https://github.com/submariner-io/submariner-charts/pull/243) should help fix this.
> 
> After merge, seeing:
> 
> ```
> [18:28:00.299] [shipyard]$ [cluster1] command kubectl --context=cluster1 -n submariner-k8s-broker get secrets submariner-k8s-broker-client-token -o jsonpath={.data['ca\.crt']}
> Error from server (NotFound): secrets "submariner-k8s-broker-client-token" not found
> ```

Our Helm tests deploy the “latest” chart, which is the latest _released_ chart, currently 0.12.2. The fixed chart is deployed as 0.13.0-rc2 which won’t be picked up automatically…

skitt committed:
Allow in-development charts

helm install installs the latest released chart by default; this
excludes pre-release charts, e.g. 0.13.0-rc2. Adding the --devel flag
allows pre-release charts to be installed; this is what we want for
in-development CI.

This will need to be revisited on branches.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

skitt committed:
Enable debug information when installing charts

This ensures that our logs include information such as the actual
version of the chart being installed.

Signed-off-by: Stephen Kitt <skitt@redhat.com>
